optimism/src/mips/interpreter.rs

```diff
@@ -899,7 +912,19 @@ pub fn interpret_rtype<Env: InterpreterEnv>(env: &mut Env, instr: RTypeInstructi
             env.set_next_instruction_pointer(next_instruction_pointer + Env::constant(4u32));
             return;
         }
-        RTypeInstruction::MultiplyToRegister => (),
+        RTypeInstruction::MultiplyToRegister => {
+            let rs = env.read_register(&rs);
+            let rt = env.read_register(&rt);
+            let res = {
+                // FIXME: Constrain
+                let pos = env.alloc_scratch();
+                unsafe { env.mul_signed_witness(&rs, &rt, pos) }
+            };
+            env.write_register(&rd, res);
```

Comment:
Do you have the spec for MUL?
I would have expected to write in HI/LO.

Reply:
You're thinking of `mult`, it's a different instruction.

Reply:
Oh no, *I'm* thinking of the wrong instruction. Lemme fix.

Reply:
Actually, wrong again. This `MultiplyToRegister` is indeed correct :man_facepalming: